src/k8bit/content.md

```diff
@@ -0,0 +1,623 @@
+**TL;DR:** _Kubernetes libraries expose the Shared Informer — an efficient pattern to watch for API changes. This article teaches you how to build your own and [a real-time dashboard for Kubernetes with only client-side Javascript.](https://github.com/learnk8s/k8bit)_
+
+In Kubernetes, you can monitor changes to Pods in real-time with the `--watch` flag:
+
+```terminal|command=1|title=bash
+kubectl get pods --watch
+```
+
+The `--watch` flag is part of the Kubernetes API, and it is designed to dispatch update events incrementally.
+
+If you tried the command in the past, you might have noticed who the output is often confusing:
+
+![`kubectl get pods --watch`](watch.svg)
+
+_How many more Pods were created?_
+
+Two, but you had to parse the output a couple of times to be sure.
+
+_Why is the command not updating the output in place?_
+
+Let's dive into what happens when you execute that command.
+
+## kubectl watch
+
+Depending on what you do, `kubectl` goes through the following steps:
+
+1. Validates client-side that the current resource is valid.
+1. Uses [generators](https://kubernetes.io/docs/user-guide/kubectl-conventions/#generators) to serialise the resources (if any).
+1. [Scans the cluster APIs for the right endpoint to call](https://github.com/kubernetes/kubernetes/blob/v1.14.0/pkg/kubectl/cmd/run/run.go#L674-L686) and [assembles a client that talk to that particular version](https://github.com/kubernetes/kubernetes/blob/v1.14.0/pkg/kubectl/cmd/run/run.go#L705-L708).
+1. Authenticates with the API using your kubeconfig credentials.
+
+The first two points are skipped here, but the remaining still apply.
```

Comment:
These points are probably not really necessary and make the matter look more complicated than it is. The crucial point for the topic of this article is that `kubectl` makes a GET request to:

```
https://api-server:8443/api/v1/namespaces/my-namespace/pods?watch=1
```